versioning/scripts/dockerfiles/main.go

```diff
@@ -101,13 +102,16 @@ func findFilesToCopy(templateDir string, callback func(path string, fileInfo os.
 }
 
 func copyFiles(version versions.Version, templateDir string) {
-	findFilesToCopy(templateDir, func(path string, fileInfo os.FileInfo) {
-		data, err := ioutil.ReadFile(filepath.Join(templateDir, path))
+	findFilesToCopy(templateDir, func(filePath string, fileInfo os.FileInfo) {
+		data, err := ioutil.ReadFile(filepath.Join(templateDir, filePath))
 		check(err)
 
-		target := filepath.Join(version.Dir, path)
+		target := filepath.Join(version.Dir, filePath)
 		// Delete first to make sure file is created with the right mode.
 		deleteIfFileExists(target)
+		// Create nested directory structure if needed.
+		dirPath := path.Dir(target)
+		os.MkdirAll(dirPath, os.ModePerm)
```

Comment:
Why do we need to create directory twice?

Please see https://github.com/ovk6/runtimes-common/blob/6a52810179b21698276ab1cb2c12c44f680f25c0/versioning/scripts/dockerfiles/main.go#L259.

Reply:
Yes, because in current configuration it was not providing nested directory structure.

Eg. in case there is nested directory:
```
templates/Dockerfile.template
templates/configs/xxx.conf
```

it was failing if directory `configs` was not created in target directory.

```
$ dockerfiles
panic: open 7/debian9/7.1/configs/xxx.conf: no such file or directory
```

Reply:
It make sense, but this the directory will be created always. I think we wanted to use the `--add_directories` flag or rather `--create_directories`?

Reply:
Yes, but I think it is a bug in original software. This directory structure should be copied even if we don't want to create new directories for new versions.

Reply:
Please correct me if I'm wrong. Will this create the directory contains `version.Dir` value even if `--add_directories` flag is `false`?